base/inference.jl

```diff
@@ -2650,16 +2760,30 @@ function substitute!(e::ANY, na, argexprs, spvals, offset)
         end
     end
     if isa(e, NewvarNode)
-        return NewvarNode(substitute!(e.slot, na, argexprs, spvals, offset))
+        return NewvarNode(substitute!(e.slot, na, argexprs, spsig, spvals, offset))
     end
     if isa(e, Expr)
         e = e::Expr
         head = e.head
         if head === :static_parameter
             return spvals[e.args[1]]
+        elseif head === :foreigncall
+            for i = 1:length(e.args)
+                if i == 2
+                    e.args[2] = ccall(:jl_instantiate_type_in_env, Any, (Any, Any, Ptr{Any}), e.args[2], spsig, spvals)
```

Comment:
Does this mean these types only get instantiated when the code is inlined somewhere?